### PR TITLE
build: use whiskers

### PR DESCRIPTION
**Pull request by uncenter**
Adds [Whiskers](https://github.com/catppuccin/toolbox/tree/main/whiskers), Catppuccin's internal templating tool, to build the themes. Instead of editing the `.json5` files directly, edit the `misskey.tera` file and then run `just build` to build the output themes. You will need Whiskers (linked above) and *optionally* [Just](https://github.com/casey/just) installed.

~~Note: currently a draft since I haven't figured out a solution for the UUIDs.~~

### CODE REVIEW

**Comment by uncenter** [comment]
I've reverted quite a few formatting changes that I had made to keep the diff minimal - so minimal in fact, there is no diff anymore. I'll PR these small changes in later but I wanted to make sure I demonstrated that this new template didn't break anything!

File: misskey.tera [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+whiskers:
+  version: "2.3.0"
+  matrix:
+   - accent
+   - flavor
+  filename: "src/themes/{{ flavor.identifier }}/catppuccin-{{ flavor.identifier }}-{{ accent }}.json5"
```
Comment (uncenter) [inline misskey.tera:7]:
```suggestion
  filename: "themes/{{ flavor.identifier }}/catppuccin-{{ flavor.identifier }}-{{ accent }}.json5"
```
I'd like to change it to this after this PR, keeping it the same for a minimal diff for now.

File: misskey.tera [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
... '#{{ palette.overlay0.hex }}',        inputBorder: 'rgba(255, 255, 255, 0.1)',        panelBorder: 'solid 1px var(--divider)',        switchOffBg: 'rgba(255, 255, 255, 0.1)',        switchOffFg: '@fg',        accentDarken: ':darken<10<@accent',        acrylicPanel: ':alpha<0.5<@panel',        navIndicator: '@indicator',        windowHeader: ':alpha<0.85<@panel',        accentLighten: ':lighten<10<@accent',        buttonHoverBg: 'rgba(255, 255, 255, 0.1)',        driveFolderBg: ':alpha<0.3<@accent',        fgHighlighted: ':lighten<3<@fg',        fgTransparent: ':alpha<0.5<@fg',        panelHeaderBg: ':lighten<3<@panel',        panelHeaderFg: '@fg',        buttonGradateA: '@accent',        buttonGradateB: ':hue<20<@accent',        htmlThemeColor: '@bg',        panelHighlight: ':lighten<3<@panel',        listItemHoverBg: 'rgba(255, 255, 255, 0.03)',        scrollbarHandle: 'rgba(255, 255, 255, 0.2)',        inputBorderHover: 'rgba(255, 255, 255, 0.2)',        wallpaperOverlay: 'rgba(0, 0, 0, 0.5)',        fgTransparentWeak: ':alpha<0.75<@fg',        panelHeaderDivider: 'rgba(0, 0, 0, 0)',        scrollbarHandleHover: 'rgba(255, 255, 255, 0.4)',    },    author: 'somebody ¯\_(ツ)_/¯',}
```
Comment (uncenter) [inline misskey.tera]:
```suggestion
{    id: '{{ read_file(path="uuids.txt") | split(pat="\n") | nth(n=flavor.order * 14 + palette[accent].order) }}',    base: '{% if flavor.dark %}dark{% else %}light{% endif %}',    name: 'Catppuccin {{ flavor.identifier }} {{ accent }}',    props: {        X2: ':darken<2<@panel',        X3: 'rgba(255, 255, 255, 0.05)',        X4: 'rgba(255, 255, 255, 0.1)',        X5: 'rgba(255, 255, 255, 0.05)',        X6: 'rgba(255, 255, 255, 0.15)',        X7: 'rgba(255, 255, 255, 0.05)',        X8: ':lighten<5<@accent',        X9: ':darken<5<@accent',        bg: '#{{ palette.crust.hex }}',        fg: '#{{ palette.text.hex }}',        X10: ':alpha<0.4<@accent',        X11: 'rgba(0, 0, 0, 0.3)',        X12: 'rgba(255, 255, 255, 0.1)',        X13: 'rgba(255, 255, 255, 0.15)',        X14: ':alpha<0.5<@navBg',        X15: ':alpha<0<@panel',        X16: ':alpha<0.7<@panel',        X17: ':alpha<0.8<@bg',        cwBg: '#{{ palette.surface1.hex }}',        cwFg: '#{{ palette.subtext1.hex }}',        link: '#{{ palette.blue.hex }}',        warn: '#{{ palette.peach.hex }}',        badge: '#{{ palette.blue.hex }}',        error: '#{{ palette.red.hex }}',        focus: ':alpha<0.3<@accent',        navBg: '@panel',        navFg: '@fg',        panel: ':lighten<3<@bg',        popup: ':lighten<3<@panel',        accent: '#{{ palette[accent].hex }}',        header: ':alpha<0.7<@panel',        infoBg: '#{{ palette.surface0.hex }}',        infoFg: '#{{ palette.subtext0.hex }}',        renote: '#{{ palette.blue.hex }}',        shadow: 'rgba(0, 0, 0, 0.3)',        divider: 'rgba(255, 255, 255, 0.1)',        hashtag: '#{{ palette.sapphire.hex }}',        mention: '@accent',        modalBg: 'rgba(0, 0, 0, 0.5)',        success: '#{{ palette.green.hex }}',        buttonBg: 'rgba(255, 255, 255, 0.05)',        switchBg: 'rgba(255, 255, 255, 0.15)',        acrylicBg: ':alpha<0.5<@bg',        cwHoverBg: '#{{ palette.surface2.hex }}',        indicator: '@accent',        mentionMe: '@mention',        messageBg: '@bg',        navActive: '@accent',        accentedBg: ':alpha<0.15<@accent',        codeNumber: '#{{ palette.green.hex }}',        codeString: '#{{ palette.peach.hex }}',        fgOnAccent: '#{{ palette.base.hex }}',        infoWarnBg: '#{{ palette.surface0.hex }}',        infoWarnFg: '#{{ palette.subtext1.hex }}',        navHoverFg: ':lighten<17<@fg',        switchOnBg: '@accentedBg',        switchOnFg: '@accent',        codeBoolean: '@accent',        dateLabelFg: '@fg',        deckDivider: '#{{ palette.overlay0.hex }}',        inputBorder: 'rgba(255, 255, 255, 0.1)',        panelBorder: 'solid 1px var(--divider)',        switchOffBg: 'rgba(255, 255, 255, 0.1)',        switchOffFg: '@fg',        accentDarken: ':darken<10<@accent',        acrylicPanel: ':alpha<0.5<@panel',        navIndicator: '@indicator',        windowHeader: ':alpha<0.85<@panel',        accentLighten: ':lighten<10<@accent',        buttonHoverBg: 'rgba(255, 255, 255, 0.1)',        driveFolderBg: ':alpha<0.3<@accent',        fgHighlighted: ':lighten<3<@fg',        fgTransparent: ':alpha<0.5<@fg',        panelHeaderBg: ':lighten<3<@panel',        panelHeaderFg: '@fg',        buttonGradateA: '@accent',        buttonGradateB: ':hue<20<@accent',        htmlThemeColor: '@bg',        panelHighlight: ':lighten<3<@panel',        listItemHoverBg: 'rgba(255, 255, 255, 0.03)',        scrollbarHandle: 'rgba(255, 255, 255, 0.2)',        inputBorderHover: 'rgba(255, 255, 255, 0.2)',        wallpaperOverlay: 'rgba(0, 0, 0, 0.5)',        fgTransparentWeak: ':alpha<0.75<@fg',        panelHeaderDivider: 'rgba(0, 0, 0, 0)',        scrollbarHandleHover: 'rgba(255, 255, 255, 0.4)',    },    author: 'somebody ¯\_(ツ)_/¯',}

```
I'd like to add a trailing newline here, keeping it the same for a minimal diff for now (this is a note to self lol)

Comment (uncenter) [reply misskey.tera]: Also would be nice to add the JSON formatting back in - split this single line into multiple lines. Also, the name and author fields should be updated.

File: uuids.txt
Comment (uncenter) [inline uuids.txt:1]: Reordered to match the palette order, needed to keep the UUIDs the same in the themes themselves.

**Comment by uncenter** [comment]
In the future you should use squash commits/merges when possible 😅 stuff like this wasn't/isn't meant to go in the repositories history:

Screenshot 2024-05-28 at 08 37 06 (Arc)

**Comment by catb00mer** [comment]
oof my bad xD

**Pull request merged**